library/Backends/llvm/src/Juvix/Backends/LLVM/Pass/Types.hs

```diff
@@ -67,7 +67,10 @@ data IndexInto = IndexInto
 -- from the closure array/environment or the closure argument array/environment
 data FunctionEnvironment
   = ClosureEnvironment
-  | ArgumentEnvironemnt
+  | -- Currently unused due to how it's impossible to distinguish
+    -- between functions and closures. We should in time compile with
+    -- this, and will be needed for currying
+    ArgumentEnvironemnt
```

Comment:
Should we maybe have a `to-fix` test for currying, then?

Reply:
Probably would be good, we need to think hard about how to get currying at not cost while folloiwng the rapier technique laid out by SPJ's paper.

We really should not sacrifice much performance for this.

Reply:
Yeah, we need to think long and hard how to implement SPJ's rapier technique wtihout sacrificing much performance. Ideally we can think hard about LLVM and what would cost us speed of execution and what the best course of action is.